src/integrations/front-end/wp-robots-integration.php

```diff
@@ -148,20 +147,23 @@ protected function format_robots( $robots ) {
 	 *
 	 * @return array The filtered robots.
 	 */
-	protected function filter_robots_no_index( $robots ) {
-		if ( ! isset( $robots['noindex'] ) ) {
-			return $robots;
+	protected function filter_robots_noindex_nofollow( $robots ) {
```

Comment:
If I read this correctly, this method does a few things right now:
* Removing the `follow` robot value when a `nofollow` value is set.
	* E.g. `nofollow` --> `follow => null`; 
* Removing the `index` robot value when a `noindex` value is set.
	* E.g. `noindex` --> `index => null`; 
* Filtering out any specific robot values when the page is set to `noindex`.

To adhere to the single responsibility principle I would split this up into three separate methods. Each method doing one of the things above.

Reply:
I disagree with this. The responsibility is to make the robots meta value congruent. I don't think extracting stuff make things more readable. In fact, I think it will primarily add a lot of scrolling. However, I do agree, if it's supposed to make the value congruent, it should cover all cases.